rundeck-data-models/src/main/java/org/rundeck/app/data/providers/v1/authtoken/TokenDataProvider.java

```diff
@@ -90,6 +90,15 @@ public interface TokenDataProvider extends DataProvider {
      */
     List<AuthenticationToken> findAllByUser(String userId);
 
+    /**
+     * Retrieves a List of AuthenticationToken for the user and type.
+     *
+     * @param userId of the AuthenticationTokens, format String
+     * @param type of the AuthenticationToken, format AuthTokenType
+     * @return list of AuthenticationTokens
+     */
+    List<AuthenticationToken> findAllByUserAndType(String userId, AuthTokenType type);
```

Comment:
to avoid the need to implement this new method on the pro side (for multi tenancy), we can just use default

Reply:
example
https://github.com/rundeck/rundeck/pull/9812/files#diff-472fe1c4b3ad5fad71966484ca71a4b67c23890619a926e516f07b97e6b04b0eR54